tools/site_cobble/rdl_pkg/templates/regpkg_bsv.jinja2

```diff
@@ -0,0 +1,84 @@
+
+// This is a generated file using the RDL tooling. Do not edit by hand.
+package {{ map_name|lower|to_camel_case(uppercamel=True) }};
+
+import Reserved::*;
+import RegCommon::*;
+
+{# Loop registers building out BSV packages #}
+{% for register in registers %}
+{% set reserved_cntr = namespace(value=0) %}
+// Register {{register.name}} definitions
+typedef struct {
+    {% for field in register.fields %}
+        {% if isinstance(field, ReservedField) %}
+            {% set field_type = "ReservedZero#({})".format(field.width) %}
+            {% set name = register.format_field_name("zeros{}".format(reserved_cntr.value)) %}
+            {% set reserved_cntr.value = reserved_cntr.value + 1 %}
+        {% else %}
+            {% set field_type = "Bit#({})".format(field.width) %}
+            {% set name =  register.format_field_name(field.name)|lower %}
+        {{ "{:<18}".format(field_type)}} {{ name }};  // bit {{field.bitslice_str()}}
+        {% endif %}
+    
+    {% endfor %}
+{% set reg_name_camel = register.name|lower|to_camel_case(uppercamel=True) %}
+} {{ reg_name_camel }} deriving (Eq, FShow);
+// Register offsets
+Integer {{register.name|lower|to_camel_case}}Offset = {{register.offset}};
+// Field mask definitions
+    {% for field in register.fields %}
+        {% if not isinstance(field, ReservedField) %}
+    Bit#({{register.width}}) {{register.name|lower|to_camel_case}}{{register.format_field_name(field.name)|lower|to_camel_case(uppercamel=True)}} = 'h{{field.mask}};
+        {% endif %}
+    {% endfor %}
+// Register {{register.name}} custom type-classes
+instance Bits#({{reg_name_camel}}, {{register.width}});
+    function Bit#({{register.width}}) pack ({{reg_name_camel}} r);
+        Bit#({{register.width}}) bts =  'h00;
+        {% for field in register.fields %}
+        {% if not isinstance(field, ReservedField) %}
+        bts[{{field.bitslice_str()}}] = r.{{register.format_field_name(field.name).strip()|lower}};
+        {% endif %}
+        {% endfor %}
+        return bts;
+    endfunction: pack
+    function {{reg_name_camel}} unpack (Bit#({{register.width}}) b);
+        let r = {{reg_name_camel}} {
+        {% for field in register.fields %}
+        {% if not isinstance(field, ReservedField) %}
+        {{register.format_field_name(field.name).strip()|lower}}: b[{{field.bitslice_str()}}] {{ ", " if not loop.last else "" }}
```

Comment:
This is the problematic logic in the bug I raised. In the case where the "last" (0th) bit is unnamed (thus reserved), it will generate a comma after the last user named field, but then never generate the final field assignment without a comma in the struct, which is a syntax error in BSV.
```
    let r = Control {
        en: b[7] , 
        ien: b[6] , <-- this comma is a syntax error
        };
```

Reply:
fixed by : https://github.com/oxidecomputer/cobalt/pull/15/commits/70e2118f88253241a204a0a3d7bfe7238a5a3e2c